go/database/mpt/verification_test.go

```diff
@@ -412,6 +417,144 @@ func TestVerification_ValueNodeHashModificationIsDetected(t *testing.T) {
 		}
 	})
 }
+
+func TestVerification_MissingCodeHashInCodeFileIsDetected(t *testing.T) {
+	runVerificationTest(t, func(t *testing.T, dir string, config MptConfig, roots []Root) {
+		err := VerifyMptState(dir, config, roots, NilVerificationObserver{})
```

Comment:
I would appreciate if those tests could be self-contained, such that everything one needs to know for the test is present in the test.

In this case, the test checks for a missing code, but this scenario is not set up in the test. A reader needs to know that there is a code expected by the created state that by default is not there.

Better: let the `runVerificationTest` function create a correct code file and explicitly remove the code which's absence should be detected.

In this case, please also add a test case where the verification for the state created by `runVerificationTest` passes the verification test.

Reply:
This was the design I went with these tests but @kjezek did not like it so we changed it. @kjezek wdyt

Reply:
I am fine with Herbert's comment. No problem to change it.

I think what Herbert asks for is not exactly reverting to what you had before. The previous version had a problem that it customised nodes in some tests (it is what Herbert asks), but in other tests it relied on empty code hashes that were the default.  If the tests are consistent, i.e. customisation is happening everywhere, I am fine with it.